gdpr/loading.py

```diff
@@ -1,57 +1,162 @@
-import six
-
-from collections import OrderedDict
-
+from collections import OrderedDict, _OrderedDictItemsView, _OrderedDictKeysView, _OrderedDictValuesView
 from importlib import import_module
+from typing import Any, Generic, Iterator, Optional, TYPE_CHECKING, Type, TypeVar, Union
 
 from django.apps import apps
 from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db.models import Model
 from django.utils.encoding import force_text
 
-from is_core.utils import str_to_class
+from .utils import str_to_class
 
+if TYPE_CHECKING:
+    from gdpr.anonymizers import ModelAnonymizer
+    from gdpr.purposes import AbstractPurpose
 
-class AppAnonymizerLoader:
-    """
-    Anonymizers loader that interates over all applications and is looks for anonymizers.py file that should contain
-    anonymizer classes.
-    """
 
-    def import_anonymizers(self):
+class BaseLoader:
+    """Base class for all loaders."""
+
+    def import_modules(self) -> None:
+        raise NotImplementedError()
+
+
+class AppLoader(BaseLoader):
+    """Scan all installed apps for `module_name` module."""
+    module_name: str
+
+    def import_modules(self) -> None:
         for app in apps.get_app_configs():
+            if app.name == "gdpr":
+                continue
             try:
-                import_module('{}.anonymizers'.format(app.name))
+                import_module(f'{app.name}.{self.module_name}')
             except ImportError as ex:
-                if ((six.PY2 and force_text(ex) != 'No module named anonymizers') or
-                        (six.PY3 and force_text(ex) != 'No module named \'{}.anonymizers\''.format(app.name))):
+                if force_text(ex) != f'No module named \'{app.name}.{self.module_name}\'':
                     raise ex
 
 
-class AnonymizersRegister:
-    """
-    Register is storage for found anonymizer classes.
-    """
+class SettingsListLoader(BaseLoader):
+    """Import all modules from list `list_name` in settings."""
+    list_name: str
+
+    def import_modules(self):
+        if not hasattr(settings, self.list_name):
+            raise ImproperlyConfigured(f"settings.{self.list_name} not found.")
+        modules_list = getattr(settings, self.list_name)
+        if type(modules_list) is str:
+            import_module(modules_list)
+        elif type(modules_list) in [list, tuple]:
+            for i in modules_list:
+                import_module(i)
```

Comment:
jen taková obecná, proč by to někdo chtěl ještě zanořovat?

Reply:
@matllubos Tam není zanoření to je aby v settings mohlo být:

```python
GDPR_ANONYMIZERS_LIST = 'comments.special_anonymizers'
```

nebo 

```python
GDPR_ANONYMIZERS_LIST = [
    'comments.special_anonymizers',
    'orders.anonymizers'
]
```

Ale tam je spíš ta první varianta blbost to je pravda.

Reply:
odebráno